iam.tf

```diff
@@ -1,5 +1,5 @@
 data "aws_iam_policy_document" "this" {
-  count = var.irsa_policy == null && !var.irsa_assume_role_enabled ? 1 : 0
+  count =  var.enabled && var.irsa_policy == null ? 1 : 0
```

Comment:
I am giving this another thought and I think we should add `var.irsa_policy_enabled != false` as well IMHO. This should prevent adding this source when irsa_policy_enabled is explicitly disabled.